internal/keepers/service.go

```diff
@@ -237,27 +154,30 @@ func (s *onDemandUpkeepService) parallelCheck(ctx context.Context, keys []types.
 	// go through keys and check the cache first
 	// if an item doesn't exist on the cache, send the items to the worker threads
 	for _, key := range keys {
-		// get reported status and continue if found
-		id, err := s.registry.IdentifierFromKey(key)
-		if err != nil {
-			return samples.Values(), err
+		add := true
+		for _, filter := range filters {
+			if !filter(key) {
+				add = false
+				break
+			}
 		}
-		state, stateCached := s.stateCache.Get(string(id))
-		if stateCached && state == types.Reported {
-			cacheHits++
+
+		if !add {
 			continue
 		}
 
-		// skip if reported
+		// no RPC lookups need to be done if a result has already been cached
 		result, cached := s.cache.Get(string(key))
 		if cached {
 			cacheHits++
-			if result.State == types.Perform {
+			if result.State == types.Eligible {
```

Comment:
not sure why we need this special logic if cached. this should be applicable in either case?

Reply:
All results should be cached, eligible or not.